test/suite/test_checkpoint04.py

```diff
@@ -68,63 +68,70 @@ def get_stat(self, stat):
 
     def test_checkpoint_stats(self):
         nrows = 100
-        ntables = 5
-
-        # Create many tables and perform many updates so our checkpoint stats are populated.
-        value = "wired" * 100
-        tables = self.create_tables(ntables)
-        for uri, ds in tables.items():
-            self.add_updates(uri, ds, value, nrows)
-
-        # Perform a checkpoint.
-        self.session.checkpoint()
-
-        # Update the tables.
-        value = "tiger" * 100
-        tables = self.create_tables(ntables)
-        for uri, ds in tables.items():
-            self.add_updates(uri, ds, value, nrows)
-
-        # Perform a checkpoint.
-        self.session.checkpoint()
-
-        # Check the statistics.
-        # Set them into a variable so that we can print them all out. We've had a failure
-        # on Windows that is very difficult to reproduce so collect what info we can.
-        num_ckpt = self.get_stat(stat.conn.txn_checkpoint)
-        self.pr('txn_checkpoint, number of checkpoints ' + str(num_ckpt))
-        running = self.get_stat(stat.conn.txn_checkpoint_running)
-        self.pr('txn_checkpoint_running ' + str(running))
-        prep_running = self.get_stat(stat.conn.txn_checkpoint_prep_running)
-        self.pr('txn_checkpoint_prep_running ' + str(prep_running))
-
-        prep_min = self.get_stat(stat.conn.txn_checkpoint_prep_min)
-        self.pr('txn_checkpoint_prep_min ' + str(prep_min))
-        time_min = self.get_stat(stat.conn.txn_checkpoint_time_min)
-        self.pr('txn_checkpoint_time_min ' + str(time_min))
-
-        prep_max = self.get_stat(stat.conn.txn_checkpoint_prep_max)
-        self.pr('txn_checkpoint_prep_max ' + str(prep_max))
-        time_max = self.get_stat(stat.conn.txn_checkpoint_time_max)
-        self.pr('txn_checkpoint_time_max ' + str(time_max))
-
-        prep_recent = self.get_stat(stat.conn.txn_checkpoint_prep_recent)
-        self.pr('txn_checkpoint_prep_recent ' + str(prep_recent))
-        time_recent = self.get_stat(stat.conn.txn_checkpoint_time_recent)
-        self.pr('txn_checkpoint_time_recent ' + str(time_recent))
-
-        prep_total = self.get_stat(stat.conn.txn_checkpoint_prep_total)
-        self.pr('txn_checkpoint_prep_total ' + str(prep_total))
-        time_total = self.get_stat(stat.conn.txn_checkpoint_time_total)
-        self.pr('txn_checkpoint_time_total ' + str(time_total))
-
-        self.assertEqual(num_ckpt, 2)
-        self.assertEqual(running, 0)
-        self.assertEqual(prep_running, 0)
-        self.assertLess(prep_min, time_min)
-        self.assertLess(prep_max, time_max)
-        self.assertLess(prep_recent, time_recent)
-        self.assertLess(prep_total, time_total)
+        ntables = 10
+        multiplier = 1
+
+        # Run the loop and increase the value size with each iteration until
+        # the test passes.
+        while True:
```

Comment:
My only concern is that if for some reason timings are working poorly this could loop forever. We should give it some maximum where we assert mulitplier is < 100 or something so that if we ever get stuck, we fail and notice it.

Reply:
@sueloverso, Yeah, I added an assert if the loop continues for more than 100 iterations.